grovedb/src/replication.rs

```diff
@@ -84,57 +105,78 @@ impl GroveDb {
             ));
         }
 
-        let root_app_hash = self.root_hash(Some(tx.as_ref()), grove_version).value?;
-        let (chunk_prefix, root_key, tree_type, chunk_id) =
-            utils::decode_global_chunk_id(global_chunk_id, &root_app_hash)?;
-
-        // TODO: Refactor this by writing fetch_chunk_inner (as only merk constructor
-        // and type are different)
-        let merk = self
-            .open_transactional_merk_by_prefix(
-                chunk_prefix,
-                root_key,
-                tree_type,
-                tx.as_ref(),
-                None,
-                grove_version,
-            )
-            .value
-            .map_err(|e| {
-                Error::CorruptedData(format!(
-                    "failed to open merk by prefix tx:{} with:{}",
-                    hex::encode(chunk_prefix),
-                    e
-                ))
-            })?;
-        if merk.is_empty_tree().unwrap() {
-            return Ok(vec![]);
+        let mut global_chunk_ids: Vec<Vec<u8>> = vec![];
+        let root_app_hash = self.root_hash(None, grove_version).value?;
```

Comment:
`root_hash` takes `TransactionArg` and we have an ongoing `TxRef` above, to keep them inside one transaction this `None` shall be changed to `Some(tx.as_ref())`

Reply:
good catch!